tests/framework/extensions/clusters/clusters.go

```diff
@@ -221,6 +221,102 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretN
 	return v1Cluster
 }
 
+// HardenK3SRKE2ClusterConfig is a constructor for a apisV1.Cluster object, to be used by the rancher.Client.Provisioning client.
+func HardenK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName, kubernetesVersion string, machinePools []apisV1.RKEMachinePool) *apisV1.Cluster {
```

Comment:
I'm just wondering, since we already have a Newk3sRKE2ClusterConfig.  
most of the config looks same except for a few values, should we just leverage the NewK3sRKE2Config instead of adding this function again.

Reply:
Great point. As not have to update all of our provisioning tests. Maybe HardenK3SRKE2ClusterConfig calls NewK3SRKE2ClusterConfig, and just updates machineGlobalConfigMap, and anything else. Which brings to a question I have had. Is the flow to hardened a cluster to do it after fact ie. update the nodes and clusters after they are created?

Reply:
I can definitely look into that, ideally I would like minimal changes to `cluster.go` as well to not overly-complicate it. Yeah, so the flow to harden the cluster is as follows:

1. Harden the nodes after the K3s cluster is created.
2. Update the cluster's configuration to use hardened parameters as defined in step 1.
3. Apply the changes step 2 defined to actually harden the cluster.

Reply:
Updated the function to leverage `NewK3SRKE2Config`, looks much cleaner! Going to resolve this conversation as a result.